substrate/client/chain-spec/src/lib.rs

```diff
@@ -16,19 +16,229 @@
+//!     </tr>
+//!     <tr>
+//!       <td>telemetryEndpoints</td>
+//!       <td>Optional list of <code>multi address, verbosity</code> of telemetry endpoints. The
+//! verbosity goes from 0 to 9. With 0 being the mode with the lowest verbosity.</td>
+//!     </tr>
+//!     <tr>
+//!       <td>protocolId</td>
+//!       <td>Optional networking protocol id that identifies the chain.</td>
+//!     </tr>
+//!     <tr>
+//!       <td>forkId</td>
+//!       <td>Optional fork id. Should most likely be left empty. Can be used to signal a fork on
+//! the network level when two chains have the same genesis hash.</td>
+//!     </tr>
+//!     <tr>
+//!       <td>properties</td>
+//!       <td>Custom properties. Shall be provided in the form of
+//! <code>key</code>-<code>value</code> json object.
+//!     </td>
+//!     </tr>
+//!     <tr>
+//!       <td>consensusEngine</td>
+//!       <td>Deprecated field. Should be ignored.</td>
+//!     </tr>
+//!     <tr>
+//!       <td>codeSubstitutes</td>
+//!       <td>Optional map of <code>block_number</code> to <code>wasm_code</code> More details in
+//! material to follow.</td>
+//!     </tr>
+//!     <tr>
+//!       <td>genesis</td>
+//!       <td>Defines the initial state of the runtime. More details in material to follow.</td>
+//!     </tr>
+//!     <tr>
+//!       <td>code</td>
+//!       <td>The runtime wasm code blob in hex format.</td>
+//!     </tr>
+//!   </tbody>
+//! </table>
+//!
+//! # `genesis`: Initial Runtime State
+//!
+//! All nodes in the network must build subsequent blocks upon exactly the same genesis block.
+//!
+//! The information configured in the `genesis` section of a chain specification is used to build
+//! the genesis storage, which is essential for creating the genesis block, since the block header
+//! includes the storage root hash.
+//!
+//! The `genesis` key of the chain specification definition describes the
+//! initial state of the runtime. For example, it may contain:
+//! - an initial list of funded accounts,
+//! - the administrative account that controls the sudo key,
+//! - an initial authorities set for consensus, etc.
+//!
+//! The compiled WASM blob of the runtime code is stored in the state (and therefore on the chain).
+//! As a result, the initial runtime must also be provided within the chain specification (under the
+//! `code` field).
+//!
+//! The genesis state can be represented in the formats described by [`GenesisFormat<G>`]
+//! enum. In essence, the most important formats are:
+//! <table>
+//!   <thead>
+//!     <tr>
+//!       <th>Format</th>
+//!       <th>Description</th>
+//!     </tr>
+//!   </thead>
+//!   <tbody>
+//!     <tr>
+//!       <td>
+//! 		<a href="enum.GenesisFormat.html#variant.RuntimeGenesisConfig"><code>runtime</code></a>
+//!       </td>
+//!       <td>A JSON object that provides an explicit and comprehensive representation of the
+//! <code>RuntimeGenesisConfig</code> struct. Must contain all the keys of the genesis config, no
```

Comment:
As this `RuntimeGenesisConfig` is automagically generated from frame macros a user may wonder what we are referring to here. Also maybe can be confused because it has the same name of the enum variant.

Maybe we should drop two words about what we are taking about here?